superset-frontend/packages/superset-ui-core/src/query/buildQueryObject.ts

```diff
@@ -89,6 +92,14 @@ export default function buildQueryObject<T extends QueryFormData>(
     ...extras,
     ...filterFormData,
   });
+  const isAdhocColumn = (v?: AdhocColumn | PhysicalColumn) =>
```

Comment:
This would require adding the chart controls as a dependency. We have `isPhysicalColumn` in packages/superset-ui-core/src/query/types/Column.ts. I am adding `isAdhocColumn` there too for convenience. We should probably better separate these type guards in the future and make them available everywhere.